packages/webapp/src/components/ImagePicker/index.tsx

```diff
@@ -25,12 +25,26 @@ import { ReactComponent as EditIcon } from '../../assets/images/farm-profile/edi
 import styles from './styles.module.scss';
 import FileSizeExceedModal from '../Modals/FileSizeExceedModal';
 
+export enum FileEvent {
+  CHANGE = 'change',
+  DRAG = 'drag',
+}
+
+export type OnFileUpload = (
+  e: ChangeEvent<HTMLInputElement> | DragEvent,
+  setPreviewUrl: (url: string) => void,
+  setFileSizeExceeded: (exceeded: boolean) => void,
+  event: FileEvent,
+) => Promise<void>;
+
 export type ImagePickerProps = {
-  onSelectImage: (file: File) => void;
+  onSelectImage?: (file: File) => void; // Required if `shouldGetImageUrl` is false or falsy.
```

Comment:
What do you think about enforcing what is written in the comments via types?

```ts
type CommonProps = {
  onRemoveImage: () => void;
  label?: string;
  optional?: boolean;
  defaultUrl?: string;
};

type ImageUrlUpload = CommonProps & {
  shouldGetImageUrl: true;
  onSelectImage?: never;
  onFileUpload: OnFileUpload;
};

type DirectImageUpload = CommonProps & {
  shouldGetImageUrl?: false;
  onSelectImage: (file: File) => void;
  onFileUpload?: never;
};

export type ImagePickerProps = ImageUrlUpload | DirectImageUpload;
```
I really don't know personally -- union types are so long and I dislike the `never` requirement -- but maybe being even more explicit about the two kinds of upload would be handy? (And you might already have better names in mind than these to describe them).

Reply:
This is what I wanted to do! Thank you Joyce!!